### DISCUSSION TITLE
Quickest way to calculate token usage with streaming responses

### DISCUSSION THREAD

**Discussion by pranavishekhar368** [microsoft/semantic-kernel#7198] [Q&A] [answered]
Bit of a novice user here -- accessing metadata doesn't seem to work with streaming responses so is the best + easiest alternative to go with telemetry? Or use some external package like Tiktoken?

**Comment by madsbolaris** [comment, marked as answer]
I'm assuming you're using Azure OpenAI. There's currently an open task on the Azure OpenAI SDK to add token information to the response payload for streaming since it currently doesn't exist. The best alternative is what you described (using a package like TikToken). It _is_, however, a bit difficult to get completely accurate answers though because it's difficult to tokenize the function information provided to the model.

**Reply by KurtP20** [reply]
Hi @madsbolaris - Any updates on when this will be implemented?
